### ISSUE TITLE
How to play private Vimeo videos in iPhone/Android?

### ISSUE THREAD

**Issue by aaalveee** [vimeo/VimeoNetworking#225]
Just integrated **VimeoNetworking** with my project. My videos are private, and I am able to fetch and list these private videos in my App. But I have no idea, how I can play these videos. 

I have **VIMVideo** object. I am able to get the video link & URI from this object. There is no direct way to find the actual file URL from this object. So how I am supposed to play the video?  I can't pass the VIMVideo link to a Player in the iOS system, because there is no file in that URL.

I saw repository **PlayerKit** by Vimeo, this PlayerKit directly takes a file URL. But in my case all my videos are private, and I have no idea how to get the actual file URL of these videos which I can pass into the PlayerKit. 

Please explain, how does this work? that will be really helpful for me get started. 

Simply put, How am I supposed to play videos in an iOS App from my Vimeo Business Account in which all videos are private.

**Comment by barkiiqbal** [comment]
Any luck? I am also trying to play the private video but cannot find a solution.

**Comment by jasonhawkins** [comment]
Hi @aaalveee and @barkiiqbal, thanks for the question and sorry it took so long for a response 😬

To get access to video URLs via the Vimeo API you need a Pro subscription. Otherwise you only have access to data related to the video; things like title, description, etc. With a Pro subscription you'll have access to Progressive playback files, as well as HLS and Dash streams for your own videos. To get access to _other_ people's video URLs you'll need to register as a partner, which can be done here: https://www.vimeopartnernetwork.com

You're totally right about **PlayerKit** though. That framework is meant as a way to consume a video for a given URL. 

I hope this helps! Let me know if you have any other questions and I'll try to get back to you soon 👍

**Comment by jasonhawkins** [comment]
@aaalveee 🚀🚀🚀

**Comment by gourav2311** [comment]
i am not able to play my private vedio

**Comment by jasonhawkins** [comment]
@gourav2311 Is your issue similar to the one described above? Are you getting an error of some sort, or is the JSON response not what you're expecting?

**Comment by Bivinoc** [comment]
@aaalveee , @jasonhawkins I need to play Vimeo private video which is protected by password , I tried public and private video in my Android App, but have to integrate Password protected video ,how can it possible ?

**Comment by rpassis** [comment]
Hi @Bivinoc could you please open a new issue and we'll work with you to get it solved?

**Comment by chandanrana** [comment]
@Bivinoc Did u found any solution to play video? Github example does not seem that helpful.

**Comment by sharadkk** [comment]
We have a Vimeo Pro account and want to integrate private videos in our android app since we don't want them to be embedded anywhere other than our website (visible only to our registered users) and on our app.

Issue is when we make the videos private, it works on website by allowing the domain but it doesn't works on our app. Can anyone suggestion a solution?

**Comment by surveshoeb** [comment]
@sharadkk Have you found any work around for your issue?
